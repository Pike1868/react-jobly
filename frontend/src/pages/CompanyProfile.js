import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import JoblyApi from "../api/JoblyApi";
import { Box, Container, Typography } from "@mui/material";
import JobList from "../components/JobList";

export default function CompanyProfile() {
  const [company, setCompany] = useState(null);
  const { handle } = useParams();

  useEffect(() => {
    async function getCompanyDetails() {
      try {
        const response = await JoblyApi.getCompany(handle);
        setCompany(response.company);
      } catch (error) {
        console.error("Error fetching company details:", error);
      }
    }
    getCompanyDetails();
  }, [handle]);

  if (!company) return <Box>Loading...</Box>;

  return (
    <Box>
      <Box
        sx={{
          bgcolor: "background.paper",
          pt: 8,
          pb: 6,
        }}
      >
        <Container maxWidth="sm">
          <Typography
            component="h3"
            variant="h2"
            align="center"
            color="text.primary"
            gutterBottom
          >
            {company.name}
          </Typography>
          <Typography variant="h2" component="h1" gutterBottom></Typography>
          <Typography variant="body1">{company.description}</Typography>
        </Container>
      </Box>

      <Container sx={{ py: 4 }} maxWidth="md">
        <JobList jobs={company.jobs} />
      </Container>
    </Box>
  );
}
