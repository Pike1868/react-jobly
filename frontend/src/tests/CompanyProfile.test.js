import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyProfile from "../pages/CompanyProfile";

jest.mock("../api/JoblyApi");

describe("CompanyProfile", () => {
  it("renders without crashing", () => {
    render(
      <MemoryRouter>
        <CompanyProfile />
      </MemoryRouter>
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CompanyProfile />
      </MemoryRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });
});
