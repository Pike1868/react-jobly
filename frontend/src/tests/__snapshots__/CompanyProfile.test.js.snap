// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyProfile matches snapshot 1`] = `[Function]`;
